### PR TITLE
DKIP-78 Develop BIM Worksheet Maintenance Function/Process

### PR DESCRIPTION
**Pull request by amohdsharfuddin**
Unit test for BIM Validation:

* BIM.xlsx file existence
* Image file existence
* Category match with Group Sheet
* Video URL validity
* SOTD date format

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
+    "eslintConfig": {
+        "extends": "react-app"
+    },
+    "browserslist": [
+        ">0.2%",
+        "not dead",
+        "not ie <= 11",
+        "not op_mini all"
+    ],
+    "devDependencies": {
```
Comment (amohdsharfuddin) [inline package.json:50]: The only new change is new devDepen, the jest-html-reporters package. File change must be detecting tab spacing as well.

File: src/__tests__/bim.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                Array.isArray(sotd) && new Set(sotd).size === sotd.length;
+            if (pass) {
+                return {
+                    message: () => `SOTD array is unique. No repeated dates.`,
+                    pass: true,
+                };
+            } else {
+                return {
+                    message: () =>
+                        `SOTD array is not unique. There are repeated dates.`,
```
Comment (ixxbcrl) [inline src/__tests__/bim.js]: You could mention the number of duplicates. Or better yet, print the duplicates in this.

File: src/__tests__/bim.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                    };
+                } else {
+                    return {
+                        message: () =>
+                            `BIM.xlsx does not exist. Please add the BIM into the correct directory.`,
+                        pass: false,
+                    };
+                }
+            } catch (error) {
+                console.error(error);
```
Comment (ixxbcrl) [inline src/__tests__/bim.js]: are these console.error necessary? If there is an error, it will be thrown either ways.

File: src/__tests__/bim.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+// Test for BIM.xlsx file existence
+test("Test 1: BIM.xlsx should exist at public/assets/BIM.xlsx ", () => {
+    const dirBIM = "public/assets/BIM.xlsx";
+    expect.extend({
+        fileExist(dir) {
+            try {
+                const pass = existsSync(dir);
+
+                if (pass) {
```
Comment (ixxbcrl) [inline src/__tests__/bim.js]: you can directly do:
 if (existsSync(dir))

**Pull request merged**